### PR TITLE
MULE-9903: Verbose logging for Runtime

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: core/src/main/java/org/mule/util/PropertyChecker.java [hunk 1]
```diff
@@ -14,18 +16,25 @@
...
     private final String propertyName;
+    private final String defaultValue;
     private Boolean override;
 
     public PropertyChecker(String propertyName)
     {
+        this(propertyName, FALSE.toString());
+    }
+
+    public PropertyChecker(String propertyName, String defaultValue)
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/util/PropertyChecker.java]: jdoc

File: core/src/main/java/org/mule/util/PropertyChecker.java [hunk 2]
```diff
@@ -14,18 +16,25 @@
 public class PropertyChecker
 {
     private final String propertyName;
+    private final String defaultValue;
     private Boolean override;
 
     public PropertyChecker(String propertyName)
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/util/PropertyChecker.java]: add jdoc and specify that default value will be false

File: core/src/main/java/org/mule/util/SplashScreen.java [hunk 3]
```diff
@@ -68,7 +80,31 @@ protected void doBody(String line)
...
     {
         // default reserved for mule core info
-    }    
+    }
+
+    protected void listItems(Collection<String> items, String description)
+    {
+        if (!items.isEmpty())
+        {
+            doBody(description);
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/util/SplashScreen.java]: doBody sounds very weird....

File: modules/launcher/src/main/java/org/mule/module/launcher/domain/DomainStartedSplashScreen.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.launcher.domain;
+
+import static org.mule.module.launcher.MuleFoldersUtil.getDomainLibFolder;
+import org.mule.module.launcher.artifact.ArtifactStartedSplashScreen;
+import org.mule.module.launcher.descriptor.DomainDescriptor;
+
+public class DomainStartedSplashScreen extends ArtifactStartedSplashScreen
```
Comment (marianogonzalez) [inline modules/launcher/src/main/java/org/mule/module/launcher/domain/DomainStartedSplashScreen.java]: jdoc

File: modules/launcher/src/main/java/org/mule/module/launcher/artifact/ArtifactStartedSplashScreen.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import static java.util.Arrays.asList;
+import org.mule.module.launcher.descriptor.ArtifactDescriptor;
+import org.mule.util.SplashScreen;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class ArtifactStartedSplashScreen extends SplashScreen
```
Comment (marianogonzalez) [inline modules/launcher/src/main/java/org/mule/module/launcher/artifact/ArtifactStartedSplashScreen.java]: jdoc

File: modules/launcher/src/main/java/org/mule/module/launcher/application/ApplicationStartedSplashScreen.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+package org.mule.module.launcher.application;
+
+import static org.mule.module.launcher.MuleFoldersUtil.getAppLibFolder;
+import org.mule.module.launcher.artifact.ArtifactStartedSplashScreen;
+import org.mule.module.launcher.descriptor.ApplicationDescriptor;
+import org.mule.module.launcher.plugin.PluginDescriptor;
+
+import java.util.Set;
+
+public class ApplicationStartedSplashScreen extends ArtifactStartedSplashScreen
```
Comment (marianogonzalez) [inline modules/launcher/src/main/java/org/mule/module/launcher/application/ApplicationStartedSplashScreen.java]: jdoc

File: core/src/main/java/org/mule/util/SplashScreen.java (hunk 3 above)
Comment (afelisatti) [inline core/src/main/java/org/mule/util/SplashScreen.java]: This is the current method to add the message from SplashScreen. I've modified the ArtifactSplashScreen ones to createMessage though.

**Pull request merged**